src/server/src/main/java/io/cassandrareaper/service/SegmentRunner.java

```diff
@@ -218,17 +218,27 @@ private boolean runRepair() {
           Optional.<RepairStatusHandler>fromNullable(this),
           potentialCoordinators,
           context.config.getJmxConnectionTimeoutInSeconds());
-      if (SEGMENT_RUNNERS.containsKey(segmentId)) {
+
+      if (SEGMENT_RUNNERS.putIfAbsent(segmentId, this) != null) {
         LOG.error("SegmentRunner already exists for segment with ID: {}", segmentId);
         throw new ReaperException("SegmentRunner already exists for segment with ID: " + segmentId);
       }
-      SEGMENT_RUNNERS.put(segmentId, this);
+
       String keyspace = repairUnit.getKeyspaceName();
       boolean fullRepair = !repairUnit.getIncrementalRepair();
 
       LazyInitializer<Set<String>> busyHosts = new BusyHostsInitializer(coordinator);
       if (!canRepair(segment, keyspace, coordinator, busyHosts)) {
-        postponeCurrentSegment();
+        LOG.info(
+            "Cannot run segment {} for repair {} at the moment. Will try again later",
+            segmentId,
+            segment.getRunId());
+        SEGMENT_RUNNERS.remove(segment.getId());
+        try {
+          Thread.sleep(SLEEP_TIME_AFTER_POSTPONE_IN_MS);
+        } catch (InterruptedException e) {
+          LOG.debug("Interrupted while sleeping after a segment was postponed... weird stuff...");
+        }
```

Comment:
yeah, you're right about this approach. haven't realised no version of a postpone action was required at all. good stuff!